backend/app/middlewares/loginAdminUser.js

```diff
@@ -0,0 +1,33 @@
+const authJwt = require("./authJwt");
+
+isAdmin = (req, res, next) => {
+  User.findById(req.userId).exec((err, user) => {
+    if (err) {
+      res.status(500).send({ message: err });
+      return;
+    }
+
+    Role.find(
+      {
+        _id: { $in: user.roles }
+      },
+      (err, roles) => {
+        if (err) {
+          res.status(500).send({ message: err });
+          return;
+        }
+
+        for (let i = 0; i < roles.length; i++) {
+          if (roles[i].name !== "admin") {
+            res.status(403).send({ message: "Forbidden" });
+            return;
+          }
+
+          next();
+        }
```

Comment:
`return res.status(403).send({ message: "Forbidden" });` Let's add this after the loop not in it.

Reply:
I'm not very sure I understand this bit. Should I make this '===' rather than '!==' such that the response after the loop is like the else statement?

Reply:
I'm not very sure I understand this bit. Should I make this '===' rather than '!==' such that the response after the loop is like the else statement?

Reply:
> `return res.status(403).send({ message: "Forbidden" });` Let's add this after the loop not in it.

I'm not very sure I understand this bit. Should I make this '===' rather than '!==' such that the response after the loop is like the else statement?

Reply:
> `return res.status(403).send({ message: "Forbidden" });` Let's add this after the loop not in it.

I'm not very sure I understand this bit. Should I make this '===' rather than '!==' such that the response after the loop is like the else statement?

Reply:
Yeah kinda. Give it a shot lets see.